build/configs/imxrt1020-evk/imxrt1020-evk_download.sh

```diff
@@ -18,34 +18,169 @@
 ###########################################################################
 # imxrt1020-evk_download.sh
 
+WARNING="\n########################################\n
+	WARNING:Make sure NO other application like \n
+	putty etc, is accessing ttyACM0. If download \n
+	fails, reset the board before next try!!\n
+	#################################################\n"
 
-THIS_PATH=`test -d ${0%/*} && cd ${0%/*}; pwd`
+echo -e $WARNING
+
+
+CURDIR=$(readlink -f "$0")
+CURDIR_PATH=$(dirname "$CURDIR")
 
 # When location of this script is changed, only TOP_PATH should be changed together!!!
-TOP_PATH=${THIS_PATH}/../../..
+TOP_PATH=${CURDIR_PATH}/../../..
 
 OS_PATH=${TOP_PATH}/os
 OUTBIN_PATH=${TOP_PATH}/build/output/bin
-
+TTYDEV="/dev/ttyACM0"
+TINYARA_BIN=${OUTBIN_PATH}/tinyara.bin
 CONFIG=${OS_PATH}/.config
-if [ ! -f ${CONFIG} ]; then
+SUDO=sudo
+
+if [ ! -f ${CONFIG} ]
+then
         echo "No .config file"
         exit 1
 fi
 
 source ${CONFIG}
-source ${THIS_PATH}/imxrt1020-evk_get_filename.sh
 
-IMXRT1020_USB_STRING=`mount | grep RT1020-EVK`
-FIND_WORD="RT1020-EVK"
+if [[ "${CONFIG_ARCH_BOARD_IMXRT1020_EVK}" != "y" ]]
+then
+	echo "Target is NOT IMXRT1020_EVK"
+	exit 1
+fi
+
+if [ ! -f ${TINYARA_BIN} ]; then
+	echo "missing file ${TINYARA_BIN}"
+	exit 1
+fi
+
+if fuser -s ${TTYDEV}
+then
+	echo "${TTYDEV} is used by another process, can't proceed"
+	exit 1
+fi
+
+##Bootstrap and configure memory##
+
+source ${CURDIR_PATH}/bootstrap.sh
+$SUDO ${CURDIR_PATH}/blhost -p $TTYDEV -- fill-memory 0x2000 0x04 0xc0000006
+$SUDO ${CURDIR_PATH}/blhost -p $TTYDEV -- configure-memory 0x09 0x2000
+
+
+##Utility function to erase a part of flash##
+
+#Input: Address and length
+function flash_erase()
+{
+	size_in_bytes=$(($2 * 1024))
+	${SUDO} ${BLHOST} -p ${TTYDEV} -- flash-erase-region $1 $size_in_bytes
+	echo "Successfull erased flash region from $1 of size $2 KB"
+}
+
+
+##Utility function to write a binary on a flash partition##
 
-for ELEMENT in $IMXRT1020_USB_STRING; do 
-	#echo $ELEMENT
-	if [[ "$ELEMENT" =~ "$FIND_WORD" ]]; then
-		IMXRT1020_USB_PATH=$ELEMENT
-	fi 
-done;
+#Input: Address and filepath
+function flash_write()
+{
+	${SUDO} ${BLHOST} -p ${TTYDEV} -- write-memory $1 $2
+	sleep 3
+}
 
-GET_FILENAME
-echo "Downloading ${FLASH_IMG}"
-cp ${FLASH_IMG} ${IMXRT1020_USB_PATH} || exit 1
+##Utility function to match partition name to binary name##
+
+#NOTE: Hard-coded for now
+function get_executable_name()
+{
+	case $1 in
```

Comment:
This should cover imxrt partition as well

Reply:
resolved as there is NO binary for imxrt now

Reply:
There isn't any binary for imxrt now. Hence resolved.

Reply:
We have to erase userfs region as well to support smartfs. SMARTFS does not erase that region before using.

Reply:
And we had better rename the partition name of smartfs from imxrt to userfs in all of defconfigs.

Reply:
Okay.